indigo_app/static/javascript/indigo/views/document_editor.js

```diff
@@ -99,6 +106,7 @@
     fullEdit: function(e) {
       e.preventDefault();
       this.editFragmentText(this.parent.fragment);
+      this.editStartedAt = new Date().toISOString();
```

Comment:
Do this inside editFragmentText instead, which is called both when the entire document is edited, and when the user clicks the 'quick edit' pencil to the left of a chapter/part/section etc.